scripts/gh_data_update/run.py

```diff
@@ -0,0 +1,214 @@
+	data = collection.find({})
+	counts = {}
+	for case in data:
+		location = case["Location"]["place_name"]
+		if "United States" not in location and location not in counts:
+			counts[location] = 1
+		elif "United States" not in location and location in counts:
+			counts[location] += 1
+	return counts
+
+
+def cdc_to_gh(cdc_data, gh_usa_data):
+	logging.info("Adjusting G.h data to match CDC counts")
+	for state, cdc_count in cdc_data.items():
+		if delta := cdc_count - gh_usa_data.get(state, 0):
+			add_or_remove_cases(delta, f"{state}, United States")
+	if not_in_cdc := list(set(gh_usa_data) - set(cdc_data)):
+		for state in not_in_cdc:
+			delta = -gh_usa_data.get(state, 0)
+			add_or_remove_cases(delta, f"{state}, United States")
+
+
+def who_to_gh(who_data, gh_world_data):
+	logging.info("Adjusting G.h data to match WHO counts")
+	for country, count in who_data.items():
+		if gh_name := WHO_TO_GH.get(country):
+			country = gh_name
+		if delta := count - gh_world_data.get(country, 0):
+			add_or_remove_cases(delta, country)
+	if not_in_who := list(set(gh_world_data) - set(who_data)):
+		for country in not_in_who:
+			who_count = 0
+			if country in WHO_TO_GH.values():
+				who_name = WHO_TO_GH[list(WHO_TO_GH.values()).index(country)]
+				who_count = who_data[who_name]
+			delta = who_count - gh_world_data.get(country, 0)
+			add_or_remove_cases(delta, country)
+
+
+def add_or_remove_cases(count, location):
+	if count > 0:
+		add_cases(count, location)
+	elif count < 0:
+		remove_cases(-count, location)
+
+
+def add_cases(count, location):
+	logging.info(f"Adding {count} cases for {location}")
+	collection = get_db_collection(GH_COLLECTION)
+	for _ in range(count):
+		case_dict = EMPTY_CASE_AS_DICT.copy()
+		case_dict["Location"] = location
+		case_dict["Case_status"] = "confirmed"
+		case = dict_to_case(case_dict)  # Janky way to get default values
+		case_dict = case_to_dict(case)
+		collection.insert_one(case_dict)
+
+
+def remove_cases(count, location):
+	logging.info(f"Removing {count} cases for {location}")
+	collection = get_db_collection(GH_COLLECTION)
+	for _ in range(count):
+		query_filter = {"$and": [{"Location": location}, {"Case_status": "confirmed"}]}
+		collection.find_one_and_update(query_filter, {"$set": {"Case_status": "discarded"}})
+
+
+def get_db_collection(collection: str) -> object:
+	try:
+		client = MongoClient(DB_CONNECTION)
+		database = client[DATABASE_NAME]
+		return database[collection]
+	except PyMongoError:
+		logging.exception(f"An error occurred trying to access collection {collection}")
+		raise
+
+
+def gh_data_to_s3():
+	logging.info("Storing G.h data as CSV files in S3")
+	csv_data = gh_data_to_csv()
+	store_file("latest.csv", csv_data)
+	store_file(f"{GH_FOLDER}/{TODAY}.csv", csv_data)
+
+
+def gh_data_to_csv() -> str:
+	logging.info("Converting G.h data to CSV")
+	try:
+		collection = get_db_collection(GH_COLLECTION)
+		cursor = collection.find({})
+		return cases_to_csv([case for case in cursor])
```

Comment:
I think passing data as inputs and outputs makes for a cleaner design, and avoids passing open connections between functions.

Reply:
Fine for the monkeypox use case, as we are less than 100k cases, but this will break down at large numbers as the entire list needs to be loaded into memory.